### PR TITLE
feat: add support for es2025 duplicate named capturing groups

### LINKED ISSUE #18330: Support for ES2025 RegExp duplicate named capturing groups
### Syntax name

Duplicate named capturing groups

### Syntax proposal URL

https://github.com/tc39/proposal-duplicate-named-capturing-groups

### Example code

```js
str.match(/(?<year>[0-9]{4})-[0-9]{2}|[0-9]{2}-(?<year>[0-9]{4})/)
```

### Implementation Checklist

- [ ] Ecma262 update: https://github.com/tc39/ecma262/pull/2721
- [X] ESTree update: That's not necessary.
- [x] Acorn update: https://github.com/acornjs/acorn/commit/9d041c4c950b687df86f87842f458ad72bd7b482
- [X] `eslint-visitor-keys` update: That's not necessary.
- [x] `espree` update: https://github.com/eslint/espree/pull/608
- [X] `eslint-scope` update: That's not necessary.
- [x] `regexpp` update: https://github.com/eslint-community/regexpp/pull/195
- [ ] `eslint`
  - [x] update espree: https://github.com/eslint/eslint/pull/18596
  - [ ] update regexpp: #18630

### Participation

- [X] I am willing to submit a pull request for this change.

### Additional comments

_No response_

### PR DESCRIPTION
**Pull request by ota-meshi**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[x] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

Close #18330

#### What changes did you make? (Give an overview)

Updates regexpp to support ES2025 regular expressions.
Modify the `no-useless-backreference` rule to handle duplicate named capturing groups.
Add test cases to verify that the `no-invalid-regexp` rule does not report duplicate named capturing groups.

### CODE REVIEW

File: lib/rules/no-useless-backreference.js [hunk 1]
```diff
@@ -72,11 +72,12 @@ module.exports = {
...
-            intoNegativeLookaround: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}' which is in a negative lookaround."
+            nested: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} from within that group.",
+            forward: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which appears later in the pattern.",
+            backward: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which appears before in the same lookbehind.",
+            disjunctive: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which is in another alternative.",
+            intoNegativeLookaround: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which is in a negative lookaround.",
+            someReason: "Backreference '{{ bref }}' will be ignored. It references group '{{ group }}'{{ otherGroups }} which cannot be referenced for some reason."
```
Comment (mdjermanovic) [inline lib/rules/no-useless-backreference.js:80]: If there are multiple groups, with different reasons, the message doesn't specify any of the reasons?

Comment (mdjermanovic) [reply lib/rules/no-useless-backreference.js:80]: When there are multiple different reasons, all but one must be that the referenced groups are in other alternatives?

What do you think about the following:

1. If the backreference references only groups in other alternatives, then we report that as the problem.
2. Otherwise, we report the problem for the group that is in the same alternative. Maybe also include in the message that the backreference also references groups in other alternatives.

**Comment by mdjermanovic** [comment]
We should also update this:

https://github.com/eslint/eslint/blob/1613e2e586423ec7871617aec4dce5c433f0e9f0/lib/rules/utils/regular-expressions.js#L11

File: lib/rules/no-useless-backreference.js (hunk 1 above)
Comment (ota-meshi) [reply lib/rules/no-useless-backreference.js:80]: Thank you for the review and comment. I like your idea :+1: I will change it to be like that.

Comment (ota-meshi) [reply lib/rules/no-useless-backreference.js]: I have changed this rule, could you please check it again?

**Comment by nzakas** [comment]
@mdjermanovic this is ready for another look

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!
